public/widgets/widget.js

```diff
@@ -0,0 +1,355 @@
+/* exported PaperBadgerWidget */
+/* global ActiveXObject: false */
+/**
+ * PaperBadger widget
+ * Displays all badges for a DOI or ORCID.
+ *
+ * @param {Object}    settings
+ * @param {string=}   settings.DOI
+ * @param {string=}   settings.ORCID
+ * @param {string}    settings.containerClass
+ * @param {string=}   settings.ORCIDWidgetType
+ * @param {string=}   settings.loaderText
+ * @param {string=}   settings.removeClass
+ * @param {Function=} settings.clickCallback
+ * @constructor
+ */
+var PaperBadgerWidget = function (settings) {
+  /**
+   * Widget container element
+   * @type {Element}
+   */
+  var containerElement;
+
+  /**
+   * Determines whether the ORCID widget shows the
+   * author's name or DOIs
+   * @type {string}
+   */
+  var ORCIDWidgetType = 'default';
+
+  /**
+   * String shown in the loading animation
+   * @type {string}
+   */
+  var loaderText = 'loading widget';
+
+  /**
+   * Element for the loading animation
+   * @type {Element}
+   */
+  var loaderElement;
+
+  /**
+   * Contains the interval id for the animation loop
+   * @type {Number}
+   */
+  var loaderInterval;
+
+  /**
+   * Checks the settings given to the function and starts
+   * the widget creation.
+   */
+  var construct = function () {
+    // fix for #135, all but the first forward slash of a DOI
```

Comment:
Long term, we should probably handle this within the API, but this works great for now!